%Discovery Notes(up to 4 pages, this is approx. 3000 words) Discovery
%notes report biologically interesting discoveries using computational
%techniques. 

%Application Notes (up to 2 pages; this is approx. 1300 words or 1000
%words plus one figure) Applications Notes are short descriptions of
%novel software or new algorithm implementations, databases and
%network services (web servers, and interfaces).

\documentclass{bioinfo}
\copyrightyear{}
\pubyear{}
%\firstpage{}
\usepackage{amsmath, amssymb}
\usepackage{natbib}
\usepackage{tikz}
\usepackage{pgfplots}
\pgfplotsset{width=7cm,compat=1.5.1} 
\pgfplotsset{width=7cm} 
\usepackage[
   colorlinks=true, linkcolor=red, 
   citecolor=blue, urlcolor=blue
 ]{hyperref} 

\usepackage{txfonts}
%  $$$: 
\usepackage[subscriptcorrection,txsy]{mtpro}
%\usepackage[mtphrb]{mtpams}
%\usepackage[mtpcal, mtpfrak]{mtpb}
%
\newcommand{\bA}{\textsf{\textit{A}}}
\newcommand{\bC}{\textsf{\textit{C}}}
\newcommand{\bG}{\textsf{\textit{G}}}
\newcommand{\bT}{\textsf{\textit{T}}}
%
%
% -----------------------------------------------------------
%
\begin{document}
\title[AID hotspots]{Constitution Analysis} 
 \author[I. T. Silva~\textit{et~al}]{%
     $A$\,$^{1,3,}$\footnote{to
  whom correspondence should be addressed}\ , 
     $B$\,$^2$ and
     $C$
  }

 \address{$^{1}$Laboratory of Molecular Immunology, The Rockefeller 
     University, 1230 York Avenue, New York, NY 10065\\
     $^{2}$Departamento de Computa\c{c}\~ao e Matem\'atica,
     Universidade de S\~ao Paulo. Av. Bandeirantes, 3900,
     Ribeir\~ao Preto,  CEP 14049-901, SP, Brazil%\\
%     $^{3}$National Institute of Science and Technology in Stem Cell
%     and Cell Therapy and Center for Cell Based Therapy. Rua Cat\~ao
%     Roxo, 2501, Ribeir\~ao Preto, CEP 14051-140, SP, Brazil
}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX} 
\editor{Associate Editor: XXXXXXX}

\maketitle
\begin{abstract}
  \section{Motivation:}
  \section{Results:} 
  \href{https://github.com/someone/rich}{\texttt{https://github.com/someone/rich}}.
  \section{Contact:}
  \href{mailto:someone@somewhere.world}{\tt someone@somewhere.world}%,
\end{abstract}

\section{Introduction}
The genome is targeted by a sophisticated and highly coordinated series of molecular events. Among these events, aberrant DNA methylation patterns in human malignancy \cite{pmid23326238}, somatic retrotransposition in human cancers \cite{pmid22745252}, AID-dependent chromosomal translocations \cite{pmid21962510} and HIV integration \cite{pmid22762018}, which arrives throughout DNA, are not randomly distributed but instead associated with chromosomal regions and contributes to disrupt the integrity of the genome and human disease. 

As result, these regions represents a genomic context in which are associate with multiple underlying mechanisms. The motif-based sequence analysis is the starting point to aim potential binding site of cis-regulatory elements associated. Nevertheless, the inherent low signal/noise ratio in sequence-based motif discovery is a limitation to detect a nucleic acid sequence pattern that has some biological significance \cite{pmid16284194}. Moreover, these events may recognise a structural feature, rather than a specific sequence motif. 

%Others approaches were introduced to detect functional regions using
%methods for computing sequence complexities
%\cite{pmid24533097,pmid21317142}. In these methods, the complexity is
%measured by the entropy which evaluates the randomness of DNA
%sequence. In particular, topological entropy has been applied to
%compute the complexity of introns, exons and promoter regions. Due to
%the finite sample and high-dimensionality problems, efforts aimed to
%overcome these problems are put forward
%\cite{pmid21317142}.\footnote{its: I would prefer to leave entropy out
%  unless we have a really good point we want to make} 

% Our work has some intersection whit the computation of `enrichment $p$-values' considered in  GO analysis. We may include the references \cite{HSL}, \cite{RPTP} (just one!) or any other if you know a better alternative. We may like to mention the paper by \cite{BM} because it also considers a test for enrichment (although it is restricted to ChiP-seq peaks, and somehow different). 

However, how exactly the pattern nucleotide composition could
influence the selection of target site selection are not well
understood. To further characterize at a genome-wide scale these
regions, we introduce a new method ({\it k-enrich}) to provide a
quantitative measure of the differential spectra of $k$-mers (DNA
'words' of length $k$) throughout target DNA. 


\section{Method}
Let $\mathcal A = \{\sf A, C, G, T\}$ and $\mathcal S \in \mathcal
A^\ell$, be a given specific string of length $\ell\geq
1$. %\footnote{according to wikipedia an $\ell$-mer of a string is any
    %possible substring of size $\ell$ of that string; clearly not
    %what we want --I stick to string!} 
In what follows, we describe a method to study the profile of $\mathcal S$ along a region of interest such as those defined by viral insertion or translocation breakpoint hotspots. This provides the means to asses the significance of a differently distributed profiles along two functionally defined regions. We specialise genomic regions with translocations hotspots as described by~\cite{pmid21962510}, but the scope is clearly not
restricted to this particular application.  

Let $h = \{h_1, \ldots, h_n\}$ bet a set of translocation breakpoint hotspots, namely a set of DNA segments characterized by having a substantially
high density of translocations events. Let $w$ be the length of the
longest of such segments. The segments $h_1, \ldots, h_n$ are aligned
with respect to their central base and then extended at both ends to
have length $w$. Next, consider the partition of resulting set of
segments into $k$ evenly spaced bins of length $\ell = w/k$. Denote by
$h_{ij}$, $1 \leq j \leq k$, the $j$th bin of the $i$th
segment. Consider now the set $r= \{r_1, \ldots, r_n\}$ of segments of
width $w$ that are either at the left or at the right of any one
segment in $h$. Likewise, let $r_{ij}$, $1\leq i \leq n$, $1 \leq j
\leq k$, be the matrix formed by bins of length $\ell$ that result by
partitioning  the elements of $r$. For any $j = 1, \ldots, k-1$ and $i
= 1, \ldots, n$, let $\xi_{ij}$ and $\eta_{ij}$ be the following
Bernoulli random variables
\begin{align*}
   &\xi_{ij} = %\mathbb 1_{\{\mathcal S \in h_{ij}\}} =
    \begin{cases}
     1, &\text{if } \mathcal S \in h_{ij} \text{ and } \mathcal S 
     \notin h_{i,j+1}\\
     1, &\text{if } \mathcal S \in h_{ij} \text{ and } \mathcal S 
     \in h_{i,j+1}\\
     0, &\text{otherwhise}
    \end{cases},\\
   &\eta_{ij} = %\mathbb 1_{\{\mathcal S \in h_{ij}\}} =
    \begin{cases}
     1, &\text{if } \mathcal S \in r_{ij} \text{ and } \mathcal S 
     \notin r_{i,j+1}\\
     1, &\text{if } \mathcal S \in r_{ij} \text{ and } \mathcal S 
     \in r_{i,j+1}\\
     0, &\text{otherwhise}
    \end{cases}.
\end{align*}
Set $\xi_{ik}  = 1$ if $\mathcal S \notin h_{i,k-1}$ and $\mathcal S
\in h_{i,k-1}$, and $\xi_{ik} = 0$ otherwhise. Similarly define
$\eta_{ik}$ by using the information in $r_{ik}$. Finaly, let 
\[
   \xi_j = \sum_{i=1}^n \xi_{ij}, \qquad 
   \eta_j = \sum_{i=1}^n \eta_{ij}.
\]
The variables $\xi_j$ and $\eta_j$, $1 \leq j \leq k$, count the
number of times that the string $\mathcal S$ occurs along of a hotspot
region and of a reference region respectively.  A schematic representation of  this procedure is shown in Figure~\ref{fig:methodscheme}.


\begin{figure}
 \input{figs/fig1}
 \caption{A: Input data segments $h_1, \ldots, h_n$ containing the
   occurence of a string $\mathcal S$ and reference segments $r_1,
   \ldots, r_n$. B: matrizes of counts for a
   particular realization of the random variables $\eta_{ij}$,
   $\xi_{ij}$. C: Profile distribution for the occurence of $\mathcal
   S$ along a hotspost and a reference region.} 
 \label{fig:methodscheme}
\end{figure}


The basic question we like to address is wether the distribution
profile of the string $\mathcal S$ is significatively different along
a typical hotspot region and a reference region. This may be assessed
by considering the following $2\times k$ contingency table 
\[
  \begin{bmatrix}
  \xi_1 & \xi_2 & \ldots & \xi_k\\
  \eta_1 & \eta_2 & \ldots & \eta_k
  \end{bmatrix},
\]
obtained by merging the two vectors of counts $\xi_j$ and
$\eta_j$. Provided the number of counts in each of the cells of this
table is sufficiently large, the significance of a differential
profile can be determined by using Pearson's $X^2$ statistic, which is
distributed according to a $\chi^2$ density with $k-1$ degrees of
freedom. Other alternatives for the large sample case exist, see for
instance~\cite{RC}, but we do not pursue this further here.  It is
well known that this procedure can give a poor approximation when
several cells present low counts (smaller than 10). This may be the
case in the current setting when analysing the profile distribution of
longer strings with $\ell \geq 10$ or even smaller but rarely occuring
strings. In these situations the significance for a differential
profile is more appropriately determined by using Fisher's exact test,
see for instance~\cite{A}. The computations necessary to derive a
$p$-value are not feasible because of the large number of contingency
tables that have to be considered as a reference set when $k$ is
large. The significance may however be approximately computed by
considering a permutation test using the method in \cite{P}. We found
that R's implementation via \texttt{fisher.test} takes only few
secconds for relatively large tables, for instance with $k=1000$. 

We provide examples for the two scenarios just mentioned by
considering strings formed by a single base and strings defined by
longer motivs with $\ell = 15$. The former provides an example where
the $X^2$ statistic is a appropriate and the latter one that is
amenable to the analysis with Fisher's exact test. 

\subsection{TC-Seq libraries}\label{methods:tcseqhs}
The TC-Seq datasets analyzed here are those described by \cite{pmid21962510}. These are deposited at Sequence Read Archive (SRA, \url{http://www.ncbi.nlm.nih.gov/sra}) under accession numbers SRA061477. These datasets are from three different translocation libraries: (i) a library from activated B cells infected with AID-expressing retrovirus (denoted hereafter as AID$^{rv}$), (ii) a library from AID-deficient B cells (denoted as AID$^{-/-}$) and (iii)  a library from activated B cells expressing wild-type levels (AID$^{wt}$). Three set of curately hotspots were defined from these samples: (i) 59 physiological hotspots in Ig-genes (AID$^{rv}$ and AID$^{wt}$ samples), (ii) 157 off-target hotspots (in non Ig-genes from AID$^{rv}$ and  AID$^{wt}$ samples) and iii) 34 hotspots from AID-deficient B cells (AID$-/-$).\footnote{The actuall choice of the data sets and the objective for this is not clear. I am really confused. We have 3 conditions: (i)=AID$^{rv}$, (ii)=AID$^{-/-}$=AID$^{ko}$, (iii)=AID$^{wt}$. Fine. Now, we make use the HS of these data sets: 59 of these are in Ig genes, by joining HS from (i) and (iii), that is by gathering the HS from both conditions that include Ig genes --is this right?; 157 are in non Ig genes (again data is from the (i) and (iii) conditions); and then 34 HS from the (ii) condition. What is the purpose of this: namely to see if there is any $k$-mer difference between Ig HS \& non-Ig HS? There could be another objective: are there any differences between a $k$-mer composition across (i), (ii) and (iii)? This is all not made clear and has to be explicited. What I mean is that we have to say what we are actually wanting to compare --what is the objective of all this?}


\section{Results \& Discussion}
%We have applied our method to further understand the genomic
%complexity at recurrent translocations hotspots in murine activated B
%lymphocytes. These translocations were induced by activation-induced
%cytidine deaminase (AID).

Common uses of $k$-mers include counting all distinct $k$-mers in genome and transcriptome assembly, variants detection and read error correction \cite{pmid23325618}. Here, we address the problem to detect functional regions across the entire genome by searching $k$-mer enrichment on DNA regions.

We have applied our method to further understand the genomic complexity at recurrent translocations hotspots locus induced by activation-induced cytidine deaminase (AID). Recurrent chromosomal translocations are associated with hematopoietic malignancies such as leukemia and lymphoma and with some sarcomas and carcinomas \cite{pmid20371343}.

Although AID specifically targets the immunoglobulin genes loci (\emph{IgH}, \emph{Igl} and \emph{Ig}$\lambda$), it also targets a array of non-immunoglobulin genes and how nucleotide composition could impact the formation of translocations require better understanding. In this work we examine the landscape of $k$-mers across the physiological and off-targets translocation hotspots from AID$^{wt}$, AID$^{rv}$ and AID$^{ko}$ samples.\footnote{OK, this is the sort of phrase that holds the key to what we want to show with these examples. Still, it is not clear enough. For instance, if we want to compare $k$-mer profiles in HS & non-HS regions in each (i), (ii) and (iii) as you state in this paragraph, why do we join the HS of \emph{wt} and \emph{rv}? Your phrase is OK but we need to say perhaps something more. Some further questions: 
\begin{enumerate}
\item what happens if we do not join \emph{wt} with \emph{rv} and analyse the HS \& non-HS profiles for the CG-mer? Is there any difference between these two groups?
\item Are the data in Figure 1(A) (i.e. for the \emph{ko} condition) for Ig or non-Ig or both?
\end{enumerate}}


The results obtained by analyzing the three hotspot sets (Section
\ref{methods:tcseqhs}) are presented in Figure 1 using 1-mer (A-mer, C-mer,
G-mer and T-mer) and 2-mer (CG-mer). The nucleotide composition for each
sample are remarkably different. The Figures 1D and 1F exhibit 1-mer
inrichment throughout of hotspots (C and G nucleotides goes up and A and T
nucleotides goes down, see Table S1 for $p$-values), but only in the
physiological targets (Figure 1D) the enrichment is sharply in the middle of
hotspots while in the off-targets is broad around the center of hotspots
(Figure 1F). Interestingly, when we search for 2-mer (CG-mer), off-targets
hostpots is marked by a high degree of CG-mer ($p$-value =
4.67179250921589e-136). This effect in off-targets AID is consistent with the theoretical mechanism for CpG-type Double Strand Breakage proposed in \cite{pmid19070581}, when a slippage event between the top and bottom strand would place the CpG within a loop, thereby making it vulnerable to AID.\footnote{OK, now comes the interpretation of all of this. My very imediate conclusion/questions from the data in Figure 1 are as follows.
\begin{enumerate}
\item CG-mer analysis: Fig (C) and Fig (E) show that non-IG AID breaks are indeed enriched in CG. This does not happens in the physiological target. Nice. But how to compare these two with Fig (A)? Again, is (A) for Ig HS, or not?
\item The description of the A, C, G, T-mers results (i.e. Fig (D) \& Fig (F)) is fine! And it also connects with the profile analysis of CG for IG and non-IG.
\item To supoort further the CpG theory, would it be interesting to look after longer CG repets, i.e. for $k$-mers like CGCG or CGCGCG or even CGCGCGCG?
\item How does the \emph{ko} data fits into the analysis. It is hard to tell. Maybe, we could leave this out of the paper?
\end{enumerate}}

According to Figure 1B, we can not observe specific preference across hotspots from AID$^{ko}$ sample, although occurs nucleotide composition enrichment for $A|C|G|T$-mer (see Table S1 for $p$-values).

Following these observations, these findings strongly suggest that either $C|G$-mer or $CG$-mer are markers for distinct sequence-dependent mechanisms that attract the AID under physiological and overexpressed levels of AID respectively.

\section{Conclusion}
We propose a standalone method ({\it k-enrich}) to calculate enrichment distribution of $k$-mer throughout of target DNA. We use a few examples to demonstrate that {\it k-enrich} provide a way to investigate the genomic complexity, although our method can be applied to any nominal variable data.


\bibliographystyle{natbib}
\scriptsize{
   \bibliography{rich}
\normalsize
}
\end{document}
