%Discovery Notes(up to 4 pages, this is approx. 3000 words) Discovery notes report biologically interesting discoveries using computational techniques.

%Application Notes (up to 2 pages; this is approx. 1300 words or 1000 words plus one figure) Applications Notes are short descriptions of novel software or new algorithm implementations, databases and network services (web servers, and interfaces). 

\documentclass{bioinfo}
\copyrightyear{}
\pubyear{}
%\firstpage{}
\usepackage{amsmath, amssymb}
\usepackage{natbib}
\usepackage{tikz}
\usepackage{pgfplots}
%\pgfplotsset{width=7cm,compat=1.5.1} 
\pgfplotsset{width=7cm} 
\usepackage[
   colorlinks=true, linkcolor=red, 
   citecolor=blue, urlcolor=blue
 ]{hyperref} 
 
\usepackage{txfonts}
%  $$$: 
%\usepackage[subscriptcorrection]{mtpro}
%\usepackage[mtphrb]{mtpams}
%\usepackage[mtpcal, mtpfrak]{mtpb}
%
\newcommand{\bA}{\textsf{\textit{A}}}
\newcommand{\bC}{\textsf{\textit{C}}}
\newcommand{\bG}{\textsf{\textit{G}}}
\newcommand{\bT}{\textsf{\textit{T}}}
%
%
% -----------------------------------------------------------
%
\begin{document}
\title[AID hotspots]{Constitution Analysis} 
 \author[I. T. Silva~\textit{et~al}]{%
     Israel T. Silva\,$^{1,3,}$\footnote{to
  whom correspondence should be addressed}\ , 
  and
       Rafael A. Rosales\,$^2$
  }

 \address{$^{1}$Laboratory of Molecular Immunology, The Rockefeller 
     University, 1230 York Avenue, New York, NY 10065\\
     $^{2}$Departamento de Computa\c{c}\~ao e Matem\'atica,
     Universidade de S\~ao Paulo. Av. Bandeirantes, 3900,
     Ribeir\~ao Preto,  CEP 14049-901, SP, Brazil%\\
%     $^{3}$National Institute of Science and Technology in Stem Cell
%     and Cell Therapy and Center for Cell Based Therapy. Rua Cat\~ao
%     Roxo, 2501, Ribeir\~ao Preto, CEP 14051-140, SP, Brazil
}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}

\maketitle
\begin{abstract}
  \section{Motivation:}
  \section{Results:} 
  \href{https://github.com/someone/rich}{\texttt{https://github.com/someone/rich}}.
  \section{Contact:}
  \href{mailto:someone@somewhere.world}{\tt someone@somewhere.world}%,
\end{abstract}

\section{Introduction}
The genome is targeted by a sophisticated and highly coordinated series of molecular events.  Among these events, aberrant DNA methylation patterns in human malignancy \cite{pmid23326238}, somatic retrotransposition in human cancers \cite{pmid22745252}, AID-dependent chromosomal translocations (Klein, 2011) and HIV integration (Cohn, 2014), which arrives throughout DNA, are not randomly distributed but instead associated with chromosomal regions and contributes to disrupt the integrity of the genome and human disease.

As result, these regions represents a genomic context in which are associate with multiple underlying mechanisms. The motif-based sequence analysis is the starting point to aim potential binding site of cis-regulatory elements associated. Nevertheless, the inherent low signal/noise ratio in sequence-based motif discovery is a limitation to detect a nucleic acid sequence pattern that has some biological significance. Moreover, these events may recognise a structural feature, rather than a specific sequence motif.

Others approaches were introduced to detect functional regions using methods for computing sequence complexities \cite{pmid24533097,pmid21317142}. In these methods, the complexity is measured by the entropy which evaluates the randomness of DNA sequence. In particular, topological entropy has been applied to compute the complexity of introns, exons and promoter regions. Due to the finite sample and high-dimensionality problems, efforts aimed to overcome these problems are put forward \cite{pmid21317142}.

However, how exactly the pattern nucleotide composition could influence the selection of target site selection are not well understood. To further characterize at a genome-wide scale these regions, we introduce a new method to provide a quantitative measure of the differential spectra of $k$-mers (DNA 'words' of length $k$) inside target DNA.


\input{figs/figure1}


\section{Methods}
We specialise to regions conforming to viral insertion hotspots, but the method is clearly not restricted to this application. Insertion hotspots are detected by using scan statistics as described in \cite{IRAMM}. Let $\{h_1, \ldots, h_n\}$, bet a set of insertion hotspots and let $w$ be the length (in base pairs) of the longest such segments. The coordinates of all the hotspots were extended at bot extremes such to match exactly this length. The resulting set of segments were partitioned into $k$ evenly spaced bins of length $\rho = w/k$. Denote by $h_{ij}$, $1 \leq j \leq k$, the $j$th interval of the $i$th segment. Let $\{r_1, \ldots, r_n\}$, be a set segments of width $w$, uniformly distributed along the genome but with no intersection with $h_{i}$, $1 \leq i \leq n$. Also, denote by $r_{ij}$ the $j$th interval of the $i$th segment.

%$$ 
% \xi_p = \frac{1}{nm}\frac{k}{w} 
%   \sum_{i=1}^m\sum_{j=1}^{n} 1_{\{\xi_{ij} \in I_p\}}
%$$

Let $\mathcal A = \{\sf A, C, G, T\}$ and $\mathcal S \in \mathcal A^\ell$, be a string of length $\ell \leq \rho$. Let $\xi_{ij}$ and $\eta_{ij}$ for $1 \leq i \leq n$, $1 \leq j \leq k$, be random variables defined as 
\[
   \xi_{ij} = %\mathbb 1_{\{\mathcal S \in h_{ij}\}} =
    \begin{cases}
     1, & \text{if } \mathcal S \in h_{ij}\\
     0, &\text{otherwhise}
    \end{cases}
    \quad\text{and}\quad
   \eta_{ij} = %\mathbb 1_{\{\mathcal S \in r_{ij}\}} =
       \begin{cases}
     1, &\text{if } \mathcal S \in r_{ij}\\
     0, &\text{otherwhise}
    \end{cases}.
\]
Consider the counts
\[
   X_j = \sum_{i=1}^n \xi_{ij}, \qquad 
   Y_j = \sum_{i=1}^n \eta_{ij}
\]
that is $X_j$ and $Y_j$ reccord the number of times that $\mathcal S$ occurs along each bin of a hotspot and a randomly chosen region respectively. In order to asses if the distribution of $\mathcal S$ varies differently across the bins of a hotspot region, we may consider the following test. Let
\[
  X^{2} = \sum_{j=1}^k (X_{j}-Y_{j})^2/Y_{j}.
\]
Provided the number of counts in each of the $k$ bins is sufficiently large, $X^{2}$ is a $\chi^{2}$ random variable with $k-1$ degrees of freedom. This provides then the sampling distribution to perform Pearson's $\chi^{2}$ test for a $2\times k$ contingency table, with $H_{0}$ and $H_{a}$ ...


\bibliographystyle{natbib}
\scriptsize{
   \bibliography{rich}
\normalsize
}
\end{document}
